finagle-mysql/src/main/scala/com/twitter/finagle/mysql/ConnectionInitSql.scala

```diff
@@ -0,0 +1,35 @@
+package com.twitter.finagle.mysql
+
+import com.twitter.finagle._
+import com.twitter.finagle.mysql.param.ConnectionInitRequest
+import com.twitter.util._
+
+/**
+  * A stack module that executes a request when create a connection.
+  */
+object ConnectionInitSql {
+  val Role: Stack.Role = Stack.Role("ConnectionInitSql")
+
+  private[finagle] def module: Stackable[ServiceFactory[Request, Result]] =
+    new Stack.Module1[ConnectionInitRequest, ServiceFactory[Request, Result]] {
+      val role: Stack.Role = Role
+      val description: String = "Installs a connection init sql module in the stack"
+
+      def make(in: ConnectionInitRequest, next: ServiceFactory[Request, Result]): ServiceFactory[Request, Result] = {
+        in.request match {
+          case Some(req) =>
+            next.flatMap { service =>
+              service(req).flatMap {
+                case _: OK => Future.value(service)
+                case Error(code, sqlState, message) =>
+                  service.close().flatMap { _ =>
+                    Future.exception(ServerError(code, sqlState, message))
+                  }
+                case r => Future.exception(new IllegalStateException(s"Unsupported response to a modify='$r'"))
```

Comment:
Not sure I would use an IllegalStateException here as
this is more like an unsupported response. Maybe just
use a generic exception instead?

Reply:
Currently ConnectionInitRequst accepts all kind of request, so sometimes mysql return not only `OK` but also `ResultSet`.
So I changed response handling which process Error and otherwise in https://github.com/twitter/finagle/pull/840/commits/d6532f8c6498bb8c871774180b586dfcaaa617b0.